src/bulk/common.ts

```diff
@@ -167,59 +163,48 @@ export class Batch<T = Document> {
   }
 }
 
+function generateIdMap(ids: Document[]): { [key: number]: any } {
+  const idMap: { [index: number]: any } = {};
+  for (const doc of ids ?? []) {
+    idMap[doc.index] = doc._id;
+  }
+  return idMap;
+}
 /**
  * @public
  * The result of a bulk write.
  */
 export class BulkWriteResult {
-  /** @deprecated Will be removed in 5.0 */
-  result: BulkResult;
-
-  /**
-   * Create a new BulkWriteResult instance
-   * @internal
-   */
-  constructor(bulkResult: BulkResult) {
-    this.result = bulkResult;
-  }
-
+  private result: BulkResult;
   /** Number of documents inserted. */
-  get insertedCount(): number {
-    return this.result.nInserted ?? 0;
-  }
+  insertedCount: number;
   /** Number of documents matched for update. */
-  get matchedCount(): number {
-    return this.result.nMatched ?? 0;
-  }
+  matchedCount: number;
   /** Number of documents modified. */
-  get modifiedCount(): number {
-    return this.result.nModified ?? 0;
-  }
+  modifiedCount: number;
   /** Number of documents deleted. */
-  get deletedCount(): number {
-    return this.result.nRemoved ?? 0;
-  }
+  deletedCount: number;
   /** Number of documents upserted. */
-  get upsertedCount(): number {
-    return this.result.upserted.length ?? 0;
-  }
-
+  upsertedCount: number;
   /** Upserted document generated Id's, hash key is the index of the originating operation */
-  get upsertedIds(): { [key: number]: any } {
-    const upserted: { [index: number]: any } = {};
-    for (const doc of this.result.upserted ?? []) {
-      upserted[doc.index] = doc._id;
-    }
-    return upserted;
-  }
-
+  upsertedIds: { [key: number]: any };
   /** Inserted document generated Id's, hash key is the index of the originating operation */
-  get insertedIds(): { [key: number]: any } {
-    const inserted: { [index: number]: any } = {};
-    for (const doc of this.result.insertedIds ?? []) {
-      inserted[doc.index] = doc._id;
-    }
-    return inserted;
+  insertedIds: { [key: number]: any };
+
+  /**
+   * Create a new BulkWriteResult instance
+   * @internal
+   */
+  constructor(bulkResult: BulkResult) {
+    this.result = bulkResult;
```

Comment:
Is there a reason to assign `result` here and then override it in line 207?

Reply:
The override was to set it as non-enumerable but typescript complained if I did it all in one go here. (It couldn't figure out that `Object.defineProperty` was setting `this.result`.